### PR TITLE
fix(lib/grandpa): update grandpa protocol ID

### LINKED ISSUE #2701: update grandpa protocol ID
### Description

Update GRANDPA protocol id from `/paritytech/grandpa/1` to `/{genesis_hash}/grandpa/1` as described here -> https://github.com/ChainSafe/gossamer/issues/2623#issuecomment-1185115378

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

- Update GRANDPA protocol id from `/paritytech/grandpa/1` to `/{genesis_hash}/grandpa/1` as described here -> https://github.com/ChainSafe/gossamer/issues/2623#issuecomment-1185115378

## Tests

N/A

## Issues

- Closes #2701

## Primary Reviewer

@jimjbrettj

### CODE REVIEW

File: lib/grandpa/network.go [hunk 1]
```diff
@@ -38,9 +38,10 @@ type GrandpaHandshake struct { //nolint:revive
 	Roles byte
 }
 
+// TODO: should we remove the SubProtocol mehtod from the `Message` interface?
```
Comment (qdm12) [inline lib/grandpa/network.go:41]: So.. should we? 😄 Also

```suggestion
// TODO: should we remove the SubProtocol method from the `Message` interface?
```

**Review by qdm12 — APPROVED** [review]

**Review by timwu20 — APPROVED** [review]

File: lib/grandpa/network.go [hunk 2]
```diff
@@ -15,10 +16,9 @@ import (
 	"github.com/libp2p/go-libp2p-core/protocol"
 )
 
-var (
-	grandpaID                protocol.ID = "/paritytech/grandpa/1"
-	messageID                            = network.ConsensusMsgType
-	neighbourMessageInterval             = time.Minute * 5
+const (
+	grandpaID1               = "grandpa/1"
+	neighbourMessageInterval = time.Minute * 5
```
Comment (qdm12) [inline lib/grandpa/network.go:21]: pedantic nit / out of scope: unit should be at the end


```suggestion
	neighbourMessageInterval = 5 * time.Minute
```

File: lib/grandpa/network.go [hunk 3]
```diff
@@ -39,8 +39,9 @@ type GrandpaHandshake struct { //nolint:revive
 }
 
 // SubProtocol returns the grandpa sub-protocol
+// TODO: should we remove the SubProtocol method from the `Message` interface?
```
Comment (qdm12) [inline lib/grandpa/network.go:42]: Please link to an issue or do it in this PR

File: lib/grandpa/network_test.go [hunk 4]
```diff
@@ -80,11 +80,7 @@ func TestHandleNetworkMessage(t *testing.T) {
 
 func TestSendNeighbourMessage(t *testing.T) {
 	gs, st := newTestService(t)
-	neighbourMessageInterval = time.Second
-	defer func() {
-		neighbourMessageInterval = time.Minute * 5
-	}()
-	go gs.sendNeighbourMessage()
+	go gs.sendNeighbourMessage(time.Second)
```
Comment (qdm12) [inline lib/grandpa/network_test.go:83]: shouldn't this be

```suggestion
	go gs.sendNeighbourMessage(neighbourMessageInterval)
```

**Review by qdm12 — CHANGES_REQUESTED** [review]

**Review by jimjbrettj — APPROVED** [review]
lgtm

**Review by kishansagathiya — CHANGES_REQUESTED** [review]
Can you post some relevant logs for this, like dump of substrate std output and gossamer std output?

When I look at the specs I see the protocol id to be `/dot/grandpa/1`, but `/paritytech/grandpa/1` is still supported, so things should work. We should as well try to support both.

Screenshot from 2022-07-20 14-34-11

https://spec.polkadot.network/#sect-protocols-substreams

**Comment by EclesioMeloJunior** [comment]
@kishansagathiya the protocol `/dot/grandpa/1` is not supported anymore and the protocol ID `/paritytech/grandpa/1` is marked as a legacy in substrate -> https://github.com/paritytech/substrate/blob/7b81072843e584ef2bee449081aed5c963021ee9/client/finality-grandpa/src/communication/mod.rs#L70, and there is an issue that describes the change -> https://github.com/paritytech/substrate/issues/7746. 

About the logs:

1. Substrate connects to another substrate and propagates messages using the new protocol ID
```
TRACE tokio-runtime-worker gossip: Registering peer who=12D3KooWFv664Y2BcR6J6eMdKuLsDGuZCFHVq6DT7mLyw6FFL4gp protocol=/588e9db899afbea5641b365fc750bcea90403573919f85d4fb2474d9057ac005/grandpa/1 role=Authority

TRACE tokio-runtime-worker gossip: Propagating message to=12D3KooWFv664Y2BcR6J6eMdKuLsDGuZCFHVq6DT7mLyw6FFL4gp protocol=/588e9db899afbea5641b365fc750bcea90403573919f85d4fb2474d9057ac005/grandpa/1
```

2. After a gossamer node started using the `/paritytech/grandpa/1`

```
INFO registered notifications sub-protocol /paritytech/grandpa/1	service.go:L538	pkg=network
Started listening on /ip4/127.0.0.1/tcp/7002/p2p/12D3KooWJ9TpY3zFzm8TzHpDKAdL4BKQu8HuZgn1kp7aKiQyZeF8
```

Substrate registered gossamer using the new protocol ID:

```
TRACE tokio-runtime-worker gossip: Registering peer who=12D3KooWJ9TpY3zFzm8TzHpDKAdL4BKQu8HuZgn1kp7aKiQyZeF8 protocol=/588e9db899afbea5641b365fc750bcea90403573919f85d4fb2474d9057ac005/grandpa/1 role=Authority
```

and propagate the message to gossamer using the new protocol ID as well:

```
TRACE tokio-runtime-worker gossip: Propagating message to=12D3KooWJ9TpY3zFzm8TzHpDKAdL4BKQu8HuZgn1kp7aKiQyZeF8 protocol=/588e9db899afbea5641b365fc750bcea90403573919f85d4fb2474d9057ac005/grandpa/1
```

here are the dumps -> https://gist.github.com/EclesioMeloJunior/f2271f1bc70dbf84fe7b427c4a60500d

**Review by qdm12 — APPROVED** [review]

**Review by kishansagathiya — APPROVED** [review]

**Pull request merged**